### PR TITLE
Fix contributing

### PR DESCRIPTION
**Pull request by dirkkul**
Closes https://github.com/weaviate/weaviate-python-client/issues/227 https://github.com/weaviate/weaviate-python-client/issues/226

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,46 @@
+### Setup
+
+We recommend to create a virtual environment to contribute to the client. You can install all developer dependencies using `pip install requirements.txt`.
+
```
Comment (samos123) [inline CONTRIBUTING.md:4]:
```suggestion
Run the following to create a virtualenvironment  and install dependencies:
```shell
python3 -m venv .venv
source .venv/bin/activate
pip install -r requirements.txt
\```
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,5 +1,46 @@
+### Setup
+
+We recommend to create a virtual environment to contribute to the client. You can install all developer dependencies using `pip install requirements.txt`.
```
Comment (samos123) [inline CONTRIBUTING.md:3]: it's `pip install -r requirements.txt` the `-r` is needed afair

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+
+There are three kinds of tests:
+- Unit tests, that test individual components of the client
+- Integration tests, that test the client with a running weaviate instance
+- Mock tests, where a weaviate instance is mocked to return specific replies
+
+To run the integration tests start the weaviate instances using `./ci/start_weaviate.sh`.
+Then run all tests with
+```
+pytest integration
```
Comment (samos123) [inline CONTRIBUTING.md:17]: I just did this:
```
./ci/start_weaviate.sh
pytest integration
```

However I'm getting a lot of failures, see partial output:
```
integration/test_authentication.py:40: WeaviateStartUpError                                                            __________ test_authentication_user_pw[okta - no refresh-test@test.de-OKTA_DUMMY_CI_PW-8083-some_scope-True] __________                                                                                                                       
recwarn = WarningsRecorder(record=True), name = 'okta - no refresh', user = 'test@test.de'                             
env_variable_name = 'OKTA_DUMMY_CI_PW', port = '8083', scope = 'some_scope', warning = True      
```

So looks like some step is missing?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,5 +1,46 @@
...
+pytest mock_tests
+pytest test
+```
+
+### Linting
+
 We use the following tools to ensure a high code quality:
 - black (formatter)
 - flake8 with plugins
```
Comment (samos123) [inline CONTRIBUTING.md:26]: how to run these locally without configuring precommit hook? what Flake8 plugins are required?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,5 +1,46 @@
...
+
+- The main branch is what is released and developed currently.
+- You can create a feature-branch that is named: feature/YOUR-FEATURE-NAME.
+- Your feature branch always has the main branch as a starting point.
+- When you are done with your feature you should create a pull request into the main branch.
+- The main branch is protected.
+
+### Contributor License Agreement
+
+Contributions to Weaviate Go client must be accompanied by a Contributor License Agreement. You (or your employer) retain the copyright to your contribution; this simply gives us permission to use and redistribute your contributions as part of Weaviate Go client. Go to [this page](https://www.semi.technology/playbooks/misc/contributor-license-agreement.html) to read the current agreement.
```
Comment (samos123) [inline CONTRIBUTING.md:40]: it mentions "Weaviate Go client" however this is the python client

**Review by samos123 — CHANGES_REQUESTED** [review]
this is great and the kind of info that would help me out making my first contributions to the python client! I had a few suggestions to improve it further

File: CONTRIBUTING.md (hunk 3 above)
Comment (dirkkul) [reply CONTRIBUTING.md:17]: Could you double check in docker if all containers are running?

File: CONTRIBUTING.md (hunk 5 above)
Comment (dirkkul) [reply CONTRIBUTING.md]: thanks, copy&paste :)

File: CONTRIBUTING.md (hunk 4 above)
Comment (dirkkul) [reply CONTRIBUTING.md]: Added (but I'd really recomment using pre-commits)

> Flake8 plugins are required

They are part of the requirements.txt file, so no furhter action is needed here.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dirkkul) [reply CONTRIBUTING.md:15]: added thanks!

**Review by StefanBogdan — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (dirkkul) [reply CONTRIBUTING.md:36]: @databyjp confirmed that the steps are working for a new setup, so I'll merge this. Just mesasge me on slack if you have further problems :)

**Pull request merged**